crates/rattler_installs_packages/src/index/http.rs

```diff
@@ -299,27 +327,58 @@ async fn fill_cache_async(
     mut body: impl Stream<Item = reqwest::Result<Bytes>> + Send + Unpin,
     handle: FileLock,
 ) -> Result<impl Read + Seek, std::io::Error> {
-    let mut cache_writer = handle.begin()?;
+    let cache_writer = handle.begin()?;
+    let mut buf_cache_writer = BufWriter::new(cache_writer);
+
+    let bom_written_position =
+        write_cache_bom(&mut buf_cache_writer, CACHE_BOM, CURRENT_VERSION).unwrap();
+
+    // We need to save struct size because we keep cache in this way:
+    // headers_struct + body
+    //
+    // When reading using `BufReader` and serializing using `ciborium`,
+    // we don't know anymore what was the final position of the struct and we
+    // can't slice and return only the body.
+    // To overcome this, we record struct size at the start of cache, together with BOM
+    // which we later will use to seek at it and return the body.
+    // Example of stored cache:
+    // [BOM][VERSION][HEADERS_STRUCT_SIZE][HEADERS][BODY]
```

Comment:
```suggestion
    // We need to save the struct size because we keep cache:
    // headers_struct + body
    //
    // When reading using `BufReader` and serializing using `ciborium`,
    // we don't know what was the final position of the struct and we
    // can't slice and return only the body, because we do not know where to start.
    // To overcome this, we record struct size at the start of cache, together with BOM
    // which we later will use to seek at it and return the body.
    // Example of stored cache:
    // [BOM][VERSION][HEADERS_STRUCT_SIZE][HEADERS][BODY]
```